app/src/main/kotlin/io/homeassistant/companion/android/controls/HaControlsProviderService.kt

```diff
@@ -249,7 +250,15 @@ class HaControlsProviderService : ControlsProviderService() {
         webSocketScope: CoroutineScope,
         subscriber: Flow.Subscriber<in Control>
     ) {
-        if (serverManager.getServer(serverId) == null) {
+        val serverCount = serverManager.defaultServers.size
+        val server = serverManager.getServer(serverId)
+
+        var serverName: String? = null
+        if (server != null && serverCount > 1) {
+            serverName = server.friendlyName
+        }
```

Comment:
In the scope of this PR I understand why we need this. But imagine opening this function in 6month will you or someone else understand why we do this? The function is already huge it's quite hard to follow. I think that you should add a small comment saying why you are doing this.

Reply:
Thanks for the feedback. I've addressed both points in the latest commit.